packages/react-link/src/components/Link/useLink.ts

```diff
@@ -1,27 +1,29 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { getNativeElementProps } from '@fluentui/react-utilities';
 import { useLinkState } from './useLinkState';
-import type { LinkProps, LinkState } from './Link.types';
+import type { LinkProps, LinkSlots, LinkState } from './Link.types';
 
 /**
  * Consts listing which props are shorthand props.
  */
-export const linkShorthandProps = [];
-
-const mergeProps = makeMergeProps<LinkState>({ deepMerge: linkShorthandProps });
+export const linkShorthandProps: Array<keyof LinkSlots> = ['root'];
 
 /**
  * Given user props, returns state and render function for a Link.
  */
-export const useLink = (props: LinkProps, ref: React.Ref<HTMLElement>, defaultProps?: LinkProps): LinkState => {
-  const state = mergeProps(
-    {
-      ref: useMergedRefs(ref, React.useRef<HTMLElement>(null)),
-      as: props.href ? 'a' : 'button',
+export const useLink = (props: LinkProps, ref: React.Ref<HTMLElement>): LinkState => {
+  const rootAs = props.href ? 'a' : 'button';
+
+  const state: LinkState = {
+    ref,
+    ...props,
+
+    components: {
+      root: rootAs,
     },
-    defaultProps,
-    resolveShorthandProps(props, linkShorthandProps),
-  );
+
+    root: getNativeElementProps(rootAs, props),
```

Comment:
```suggestion
    root: getNativeElementProps(props.as || 'a', props),
```

This should use `props.as` and fallback (to be customizable) to `components.root` definition. This looks like too much from my POV and it could be improved to simplify. WDYT?

Reply:
Makes sense, I've changed this.